src/Symfony/Component/AssetMapper/ImportMap/ImportMapRenderer.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+/**
+ * @experimental
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class ImportMapRenderer
+{
+    public function __construct(
+        private readonly ImportMapManager $importMapManager,
+        private readonly string $charset = 'UTF-8',
+        private readonly string|false $polyfillUrl = ImportMapManager::POLYFILL_URL,
+        private readonly array $scriptAttributes = [],
+    ) {
+    }
+
+    public function render(?string $entryPoint = null): string
+    {
+        $attributeString = '';
+        foreach ($this->scriptAttributes as $name => $value) {
+            $attributeString .= ' ';
+            if ($value === true) {
+                $attributeString .= $name;
+
+                continue;
+            }
+            $attributeString .= sprintf('%s="%s"', $name, $this->escapeAttributeValue($value));
+        }
+
+        $output = <<<HTML
+            <script type="importmap"{$attributeString}>
+            {$this->importMapManager->getImportMapJson()}
+            </script>
+            HTML;
+
+        if ($this->polyfillUrl) {
+            $url = $this->escapeAttributeValue($this->polyfillUrl);
+
+            $output .= <<<HTML
+
+                <!-- ES Module Shims: Import maps polyfill for modules browsers without import maps support -->
+                <script async src="$url"$attributeString></script>
```

Comment:
should we really put the attributes on both the importmap script and the polyfill script ?

Reply:
I saw a YouTube video where DHH was demo'ing Rails and it had this ;). I kind of agree with you - but I was playing it safe.